### DISCUSSION TITLE
How to use watcher?

### DISCUSSION THREAD

**Discussion by donPuerto** [vuejs/pinia#478] [Help and Questions] [answered]
Hi, I'm getting error at end. Any idea how to use or place the code?

```js
watch(
  pinia.state,
  (state) => {
    // persist the whole state to the local storage whenever it changes
    localStorage.setItem('piniaState', JSON.stringify(state))
  },
  { deep: true }
)

```
watch() -- error: cannot fine name "watch"
![image](https://user-images.githubusercontent.com/9329005/117384880-832f7980-af16-11eb-9c10-e6ea2e32ec6f.png)

**Comment by posva** [comment, marked as answer]
you need to `import { watch } from 'vue'` or `from '@vue/composition-api` depending on what you are using.

BTW there is a  [Discord chat](https://vue-land.js.org/) to ask quick questions 🙂

**Reply by donPuerto** [reply]
Nah vue land its too crowded. Here is peaceful ;)

**Reply by donPuerto** [reply]
Hi, How can the second argument of the watcher does not work?

```js
import { defineStore } from 'pinia'
import { watch } from 'vue'

export const useAuthStore = defineStore({
  // unique id of the store across your application
  id: 'authStore',
  state: ()=> {
   return {
     users: [],
     currentUser: null
   }
  },

})

const authStore = useAuthStore()
console.log('start watch')

watch(
  authStore.$state,
  (state) => {
    console.log('Watch 2', state)  // doest not do anything
  },
  { deep: true }
)

console.log('end watch')

```

**Reply by posva** [reply]
it's `watch(() => authStore.$state)` (it's in the docs: https://pinia.esm.dev/core-concepts/state.html#watching-the-state 😉 ), only Refs can be passed directly without wrapping them in a function.
